### DISCUSSION TITLE
Issues with the Mac version of OpenGothic

### DISCUSSION THREAD

**Discussion by BenniArt** [Try/OpenGothic#406] [Q&A]
Hey, first of all I'm sorry if I make a mistake; this is my first time posting to GitHub, so please bear with me.
I just learned about your project's existence yesterday while browsing reddit and had to check it out, Gothic II being one of my favourite GOATs. Unfortunately, I ran into issues running the game on my Macbook, so I figured I'd tell you about them in hopes of finding a solution sometime in the future. First a word about the hard- and software: I'm on a 2020 M1 Macbook Air with 8GB of Ram, and I'm running Ventura 13.1 as my OS. I downloaded OpenGothic from the [Mac Source Ports](https://macsourceports.com/game/gothic2) website yesterday and used the file extractor provided there to get the files from my GoG installer of the game. I've got the Gothic II: Gold Edition. Two entirely separate issues, one of which is just a personal pet peeve and curiosum, the other an objective dealbreaker.

1. At first, I successfully installed the game according to the instructions provided on the website and it launched without issues; I was able to change settings and navigate the menus. As soon as I'd start a new game, the game would crash, 100% of the time. Taking the wildest guess it turns out that the game can't be played if one uses the german game installer provided by GoG. When I downloaded the english version of the game, it installed and played just as I anticipated. I've confirmed this by reinstalling OpenGothic, re-extracting the files several times, restarting the computer, redownloading the installer, and wiping the folders in between all of these actions, none of which helped. Changing the installer did. If that is a problem that you could solve, I would be over the moon.

2. The more serious issue came later, when I was finally able to play the game. It runs amazingly well, performance-wise, especially compared to the slideshow that is the CrossOver experience. Unfortunately, the UI is extremely small, which, I've been told, can be fixed by changing the screen's resolution in the system settings, but if that's something you could change at all, that would be great, cause I can change all kinds of settings in-game, but the box for resolution is blank. The much bigger problem, however, is the behaviour of the mouse. Basically, as soon as I use it - either touchpad or an external mouse -, the camera 'falls' behind the character and looks straight up his back, and it takes a lot of random movement and effort to get it back up into position. Of course, one can play this game by constantly pressing R, but being completely unable to look around at all makes it all a little too much.

That's about it, everything else I was in awe of, and your work is very impressive. If that's something you could maybe tackle at some point, I would be grateful, and if I can help you with any data or crash reports on my part, please let me know, and it'll be my pleasure!

Take care, all the best

**Comment by Try** [comment]
Hi, @BenniArt that quite a feedback :D !

> I downloaded OpenGothic from the [Mac Source Ports](https://macsourceports.com/game/gothic2) website yesterday

For clarity: this guys having their own fork of OpenGothic, and I can't really tell for their changes. At least some of issues might be related, since game was only tested as standalone executable (not app bundle) here.

>  2020 M1 Macbook Air

Same on for me.

> As soon as I'd start a new game, the game would crash, 100% of the time. 

Need a `log.txt` and `crash.log` file from game folder to investigate.

> Unfortunately, the UI is extremely small

Issue here is, that game-ui was make with pixel-coordinates back in days, and we do not yet have a strategy on how-to scale it.
Issue been reported already here: https://github.com/Try/OpenGothic/issues/301

> the camera 'falls' behind the character and looks straight up his back,

Can you illustrate what 'falls' mean with screen-shoot?

**Reply by BenniArt** [reply]
Oh, wow, I didn't expect an answer so quickly! ^^

I'll be happy to switch to your original version to rule out any issues with the fork, but unfortunately I'm not very experienced: I did copy-paste your installation instructions into terminal and successfully got to the point where I have the folder in my user directory, but I am unsure how to proceed from here. I tried pasting both the installation .exe from GoG and the game's .exe that I got via CrossOver into the opengothic folder, cause that's how I interpreted your 'locate the executables at OpenGothic/build/opengothic', but it doesn't seem to work. Could you please tell me how to finish the installation process? I'll report back once I get over this hurdle. Thanks!

<img width="693" alt="image" src="https://user-images.githubusercontent.com/125148058/218288310-86468601-d60a-4473-86a9-7832b242e9a1.png">

**Reply by Try** [reply]
To run the game, in command line:
```
cd /path/to/build/opengothic
./Gothic2Notr.sh -g ~/Library/Application Support/OpenGothic
```

**Reply by BenniArt** [reply]
So, I've got some great news and some bad news: With the help of a couple of more knowledgable friends I was able to get the game to run (the issue seems to have been putting the game files into the Application Support folder, so for now they reside in a build/OpenGothic/ subfolder), and it has no issues running with either the English or the German files - no difference whatsoever! So that's solved and a relief.
Unfortunately, the weird mouse behaviour remains. Here's a screenshot:
![Gothic2Notr_CameraBug](https://user-images.githubusercontent.com/125148058/218318140-98ac2079-6cea-4ef1-a9c9-cb580adbf17a.jpeg)
And here I'm trying to show it in a screen capture, hope this works for you:
https://youtu.be/HLZvXRABIPk

As you can see, I move around without touching the mouse for the first couple of seconds. As soon as I do, it drops. Any idea on how to solve this? Lowering the mouse sensitivity in the in-game menu did not help; it just makes the camera drop proportionally slower. When I move straight to the left on the mousepad, the camera goes left and down, basically.

**Reply by Try** [reply]
Thanks for video!
> As you can see, I move around without touching the mouse for the first couple of seconds.

Hm, it seems that game somehow runs in full-screen mode only partially here. Can you check if toggling fullscreen works ?
To do this use `F3` (or green button on window header)

**Reply by BenniArt** [reply]
Pressing F3 allows me to toggle between seeing the window header with the buttons and the state that you see in the screenshot above. Maybe I've got you confused by the fact that I don't autohide the menubar in fullscreen apps? When I toggle to the window header option, the system's mouse is on top of the window and I can't look around in-game, but zooming in and out works, weirdly. If I left-click and drag the mouse, the camera starts spinning out of bounds, a full 360° above and below the player character.

**Reply by Try** [reply]
> the system's mouse is on top of the window

Game suppose to grab mouse control (when in full-screen mode):
Every frame game:
1. checks how far mouse pointer is from center, measures this delta and drives camera rotation from it 
2. puts mouse back in geometric center of window

> the system's mouse is on top of the window

And on top is... at windows header kind of on-top, or on-top of window in the center?

**Reply by BenniArt** [reply]
Game grabs control of the mouse in full-screen mode:
![SCR-20230212-nml](https://user-images.githubusercontent.com/125148058/218322365-1cab8bf6-542e-4c28-9d40-f5cecc313a7f.jpeg)
I can move the mouse anywhere on screen when in window mode:
![SCR-20230212-nms](https://user-images.githubusercontent.com/125148058/218322372-cfe18993-86b1-4539-b1a8-606e1b3ef5b8.jpeg)

![SCR-20230212-nmv](https://user-images.githubusercontent.com/125148058/218322379-bd1b6716-2314-4d79-ac02-5cb9a243fbf9.jpeg)
And this is what happens when I left-click and drag in this situation, just as an example:
![SCR-20230212-nn8](https://user-images.githubusercontent.com/125148058/218322398-d349ff3d-6331-45bc-81b0-9e79bb395b48.jpeg)

**Reply by BenniArt** [reply]
I've solved it! It really was the autohide setting of the menu bar - once you activate it, the mouse behaves perfectly! What a weird thing to track down...
